SIDFactoryII/source/utils/global.cpp

```diff
@@ -0,0 +1,57 @@
+#include "global.h"
+#include "configfile.h"
+#include "foundation/base/assert.h"
+#include "foundation/platform/platform_factory.h"
+#include <string>
+#include <vector>
+
+using namespace Foundation;
+
+namespace Utility
+{
+
+	Global& Global::instance()
+	{
+		static Global instance;
+		return instance;
+	}
+
+	// TODO: return const
+	IPlatform& Global::GetPlatform() const
+	{
+		FOUNDATION_ASSERT(m_Platform != nullptr);
+		return *m_Platform;
+	}
+	// TODO: return const
+	ConfigFile& Global::GetConfig() const
+	{
+		FOUNDATION_ASSERT(m_Config != nullptr);
+		return *m_Config;
+	}
+
+	Global::Global()
+		: m_Platform(Foundation::CreatePlatform())
+	{
+
+		// Read the config file
+		std::vector<std::string> valid_configuration_sections;
+		valid_configuration_sections.push_back("default");
+		valid_configuration_sections.push_back(m_Platform->GetName());
+#ifdef _DEBUG
+		valid_configuration_sections.push_back("debug");
+#endif //
+
+		std::string config_path = m_Platform->Storage_GetConfigHomePath();
+
+		ConfigFile configFile(*m_Platform, config_path + "config.ini", valid_configuration_sections);
```

Comment:
This won't work.
Instead you have to do this:

m_Config = new ConfigFile( all_the_arguments_above );

And then in the destructor you have to delete it!

The reason it doesn't work is that the configFile instance is scope to the constructor only, and so is automatically deleted at the end of it, which means you end up having a pointer to garbage in m_Config

Reply:
Ah this must be the segmentation fault! I am not used to pointers going out of scope at all :)

Reply:
I think I did the desctructor thing, but there is also a deletePlatform method, which is probably a 1:1 leftover from refactoring... do I need to remove the deletePlatform and call delete Global from main? Will that take care of deleting mPlatform and mConfig aswell?